### PR TITLE
Bump versions to prepare for release v0.22

### PR DESCRIPTION
**Pull request by djc**
Hope this helps. Is the release process written down anywhere?

### CODE REVIEW

**Comment by cijothomas** [comment]
> Hope this helps. Is the release process written down anywhere?

no. See https://github.com/open-telemetry/opentelemetry-rust/issues/1326 tracking issue.

**Review by lalitb — APPROVED** [review]

**Comment by cijothomas** [comment]
We also need changelog updates as well, similar to https://github.com/open-telemetry/opentelemetry-rust/pull/1343/files

**Comment by djc** [comment]
Added the versions to changelogs.

**Comment by jtescher** [comment]
Added the @open-telemetry/rust-maintainers group as crates.io owners, so any member should have permissions to manage them now 👍

**Comment by hdost** [comment]
Perhaps we merge https://github.com/open-telemetry/opentelemetry-rust/pull/1553 it's a simple change but a good quality of life one.  

We'd need to rebase due to conflicts, but I think it'd be worth it.

**Comment by hdost** [comment]
Perhaps also https://github.com/open-telemetry/opentelemetry-rust/pull/1556

File: opentelemetry-jaeger/CHANGELOG.md
Comment (hdost) [inline opentelemetry-jaeger/CHANGELOG.md:5]: https://github.com/open-telemetry/opentelemetry-rust/pull/1487 

There was a bad commit and they already added v0.21.0, we'll need to rebase this to fix it.

**Review by hdost — CHANGES_REQUESTED** [review]

File: opentelemetry-jaeger-propagator/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry-jaeger-propagator/CHANGELOG.md:5]: this would be the first release of this crate. So lets keep 0.1.0

File: opentelemetry-jaeger-propagator/Cargo.toml
Comment (cijothomas) [inline opentelemetry-jaeger-propagator/Cargo.toml:3]:
```suggestion
version = "0.1.0"
```

File: opentelemetry-jaeger/CHANGELOG.md
Comment (cijothomas) [reply opentelemetry-jaeger/CHANGELOG.md:5]: lets also add a note that : This will be the last release of this crate.

File: opentelemetry-jaeger/Cargo.toml
Comment (cijothomas) [inline opentelemetry-jaeger/Cargo.toml:56]:
```suggestion
opentelemetry-jaeger-propagator = { version = "0.1", path = "../opentelemetry-jaeger-propagator" } # for doctests
```

**Review by cijothomas — CHANGES_REQUESTED** [review]
Need 2 fixes
1. opentelemetry-jaeger - need to indicate this will be the last release of the create
2. opentelemetry-jaeger-propagator - use 0.1.0 as version, given this is the very first release.

Rest looks good. Though I am unsure how we decide if the new version is just minor version bump or patch version? like 0.22.0 to 0.23.0 vs 0.22.1? Maybe this is something we'll have sort out and a clear policy going forward, not blocking/delaying this release.

**Comment by TommyCpp** [comment]
> Though I am unsure how we decide if the new version is just minor version bump or patch version? like 0.22.0 to 0.23.0 vs 0.22.1

If there is a breaking change it has to be minor version bump before 1.0. Versioning was discussed here https://github.com/open-telemetry/opentelemetry-rust/blob/main/VERSIONING.md

File: opentelemetry-jaeger/CHANGELOG.md
Comment (TommyCpp) [reply opentelemetry-jaeger/CHANGELOG.md:5]: along with the note of last relase I'd also note the propagator will move to `opentelemetry-jaeger-propagator`

**Comment by hdost** [comment]
> Need 2 fixes
> 
>     1. opentelemetry-jaeger - need to indicate this will be the last release of the create
> 
>     2. opentelemetry-jaeger-propagator - use 0.1.0 as version, given this is the very first release.
> 
> 

While I don't disagree with us deprecating it, something that never made it back into the notes of how we'll proceed with properly deprecating. 
I felt that ideally we should wait for 1.0, but that was when 1.0 seemed like it might come sooner. 
However, I do think we should allow at least 2 releases I believe before outright removing it.
This one could be one which means that 0.22.0 would then be the last release.

Things to note: 
- Jaeger is still 17% of our user base.
- Today if I go to the https://crates.io/crates/opentelemetry-jaeger there's no warning shown, which means that even if someone today went to the page they might have been surprised to find out our intentions to stop supporting.

I have created https://github.com/open-telemetry/opentelemetry-rust/pull/1560 to address this.


Related issue: https://github.com/open-telemetry/opentelemetry-rust/issues/995

File: opentelemetry-appender-log/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry-appender-log/CHANGELOG.md:5]: Given there is no breaking change here, do we need to use 0.3.0? A 0.2.1 would be more apt? 
@TommyCpp @hdost do you have thoughts on this? 

Or alternately, I'd prefer is all crates from this repo uses exact same version now, to make the whole releasing easier.
https://github.com/open-telemetry/opentelemetry-rust/pull/1555#discussion_r1493981609

Maybe stick the current practice for *this* release to unblock it quickly, and the next one onwards can adopt new versioning style, if we all agree.

Comment (hdost) [reply opentelemetry-appender-log/CHANGELOG.md:5]: Technically the bump in OpenTelemetry would be a breaking change, no?

Comment (hdost) [reply opentelemetry-appender-log/CHANGELOG.md:5]: And yea i think we mentioned in another thread about making all versions the same across components in the repo.

Comment (djc) [reply opentelemetry-appender-log/CHANGELOG.md:5]: So you can't really have both, right? Either you have semver-compatible versions in some crates or you have all the crates with the same version. Randomly skipping versions is possible but is also kinda weird for downstream consumers.

In the projects I'm involved with, we always stuck to proper semantic versioning. I think a decent compromise might be to document the version of the opentelemetry core crate that the crate is compatible with somewhere near the top of the README?

Comment (cijothomas) [reply opentelemetry-appender-log/CHANGELOG.md:5]: yes we don't need to fix this part in this PR. Will use https://github.com/open-telemetry/opentelemetry-rust/pull/1555#discussion_r1493808966 to polish the process in the future.

Comment (hdost) [reply opentelemetry-appender-log/CHANGELOG.md:5]: I'm trying to visualize this because what I am imagining is something like this. Semver only requires that you have x.y.z x is major, y minor and z patch. Since we're pre 1.0 any y can be breaking. But to my knowledge there's no requirement that for a given x,y,z that the number needs to be sequential. So going from v1 to v3 is technically acceptable if not "desirable". 

Here's what i am imagining for our purposes. Example is API/SDK represented with A and some dependent crate B. 

Today:
A - 0.21.0 -> 0.22.0
B - 0.2.1 -> 0.3.0

Same version example (Minor release)
A - 0.21.0 -> 0.22.0 -> 0.23.0
B - 0.2.1 -> 0.22.0 -> 0.23.0

Version skip 
(Same version) Patch
A - 0.21.0 -> 0.22.0 -> 0.22.1
B - 0.2.1 -> 0.22.0 -> nothing 

Next minor release 
A - 0.22.1 -> 0.23.0
B - 0.22.0 -> 0.23.0 

I can understand the initial confusion with the jump but that could be addressed in the changelog. 

And as I was mentioning before in the other thread I think it makes sense more in a pre-1.0 context because of how version resolution works.

Comment (cijothomas) [reply opentelemetry-appender-log/CHANGELOG.md:5]: @hdost are you suggesting that we do it for *this* release onwards on the *next* one? I am okay with either. If we can start with *this* release, its better.

**Comment by djc** [comment]
I'm happy to do another pass on the PR once there is a clear set of remaining issues to address.

File: opentelemetry-appender-log/CHANGELOG.md
Comment (hdost) [reply opentelemetry-appender-log/CHANGELOG.md:5]: > @hdost are you suggesting that we do it for *this* release onwards on the *next* one? I am okay with either. If we can start with *this* release, its better.

Also to clarify I don't think we should block this release on the conversation. For this one, let's keep the convention we're already doing for this one. We already have the other PR talking about release process.

Comment (cijothomas) [reply opentelemetry-appender-log/CHANGELOG.md:5]: got it. Lets continue the current release with existing convention.

File: opentelemetry-jaeger/CHANGELOG.md
Comment (cijothomas) [reply opentelemetry-jaeger/CHANGELOG.md:5]: resolving this, as this is taken care by https://github.com/open-telemetry/opentelemetry-rust/pull/1560

**Comment by cijothomas** [comment]
> I'm happy to do another pass on the PR once there is a clear set of remaining issues to address, but would prefer to hold off on doing that while some comment threads have not arrived at a resolution.

There are no open questions now. This release will follow the existing convention, and that is already followed in this PR.
Remaining issues:
1. jaeger propagator crate - use 0.1.0 as this is the first release
2. jaeger crate deprecation issue is resolved - just rebase from main is good enough.
3. resolve the merge conflict.

@TommyCpp @hdost Please check if this is good, so that @djc can continue.

**Comment by TommyCpp** [comment]
> jaeger propagator crate - use 0.1.0 as this is the first release
jaeger crate deprecation issue is resolved - just rebase from main is good enough.
resolve the merge conflict.

LGTM 👍

**Comment by hdost** [comment]
Yea let's go 🚀

File: opentelemetry-appender-log/CHANGELOG.md
Comment (djc) [reply opentelemetry-appender-log/CHANGELOG.md:5]: BTW, opentelemetry-appender-log does have opentelemetry logs appearing in its public API, so it does actually need a semver-incompatible release number.

**Comment by djc** [comment]
BTW, opentelemetry-appender-log does have opentelemetry logs appearing in its public API, so it does actually need a semver-incompatible release number.

**Review by hdost — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
Thank you!

**Comment by hdost** [comment]
@TommyCpp if you want to do the honors of the merge and release this time ?

**Comment by TommyCpp** [comment]
I want to get #1562 in before the release but yeah I can do a version release this weekend

**Comment by djc** [comment]
> > I want to get #1562 in before the release but yeah I can do a version release this weekend
> 
> Could we do the release at the earliest, without waiting for 1562? At this stage, we are not merging any other PRs , to avoid interfering with the release, and we don't want to keep other PRs blocked for long..

As long as we get #1562 in this weekend and also release, that doesn't seem like a lot of extra delay?

**Comment by cijothomas** [comment]
> > > I want to get #1562 in before the release but yeah I can do a version release this weekend
> > 
> > 
> > Could we do the release at the earliest, without waiting for 1562? At this stage, we are not merging any other PRs , to avoid interfering with the release, and we don't want to keep other PRs blocked for long..
> 
> As long as we get #1562 in this weekend and also release, that doesn't seem like a lot of extra delay?

Between today vs weekend - no diff for me.
I didn't want to as people to work on a weekend. 

(for the next release onwards, we can plan and try to get release process complete in under a day)

**Review by TommyCpp — APPROVED** [review]

**Pull request merged**
